test/SPIRV/transcoding/BuildNDRange_2.ll

```diff
@@ -0,0 +1,363 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o %t.spv.txt
+; RUN: FileCheck < %t.spv.txt %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis %t.rev.bc
+; RUN: FileCheck < %t.rev.ll %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV-DAG: BuildNDRange {{[0-9]+}} {{[0-9]+}} [[GWS:[0-9]+]] [[LWS:[0-9]+]] [[GWO:[0-9]+]]
+; CHECK-SPIRV-DAG: BuildNDRange {{[0-9]+}} {{[0-9]+}} [[GWS:[0-9]+]] [[LWS:[0-9]+]] [[GWO:[0-9]+]]
+; CHECK-SPIRV-DAG: BuildNDRange {{[0-9]+}} {{[0-9]+}} [[GWS:[0-9]+]] [[LWS:[0-9]+]] [[GWO:[0-9]+]]
+
+; CHECK-LLVM: call spir_func void @_Z10ndrange_1Djjj(%struct.ndrange_t* %ndrange,
+; CHECK-LLVM: call spir_func void @_Z10ndrange_2D
+; CHECK-LLVM: call spir_func void @_Z10ndrange_3D
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+%opencl.queue_t = type opaque
+%opencl.block = type opaque
+%struct.ndrange_t = type <{ i32, [3 x i64], [3 x i64], [3 x i64] }>
+
+@block_fn.gs = internal constant [2 x i32] [i32 64, i32 4096], align 4
+@block_fn.gs1 = internal constant [3 x i32] [i32 64, i32 64, i32 64], align 4
+
+; Function Attrs: nounwind
+define spir_func void @block_fn(i32 %level, i32 %maxGlobalWorkSize, i32 addrspace(1)* %rnd, i32 addrspace(1)* %res) #0 {
+entry:
+  %level.addr = alloca i32, align 4
```

Comment:
Please, reduce the test case to have minimal number of instructions. Preferably 3 calls of built-in function with constant arguments.